### ISSUE TITLE
Permissions - display over other apps

### ISSUE THREAD

**Issue by moesaid** [ZEGOCLOUD/zego_uikit_prebuilt_call_flutter#56]
- [x] I have read the [Getting Started](https://www.zegocloud.com/doc/callkit/Call%20Kit/main/Quick%20start/b26d509e) and [Getting Started(with call invitation)](https://www.zegocloud.com/doc/callkit/Call%20Kit/main/Quick%20start%20(with%20call%20invitation)/eb1d3c42)
- [x] I have already searched for the same problem

## Environment

| Technology            | Version       |
|-----------------------| ------------- |
| Flutter version       |       3.24.0        |
| Plugin version        |      4.14.3         |
| Android version       |      vanillaIceCream         |
| iOS version           |               |
| macOS version         |               |
| Xcode version         |               |
| Google Chrome version |               |

Device information: 
emulator

## Description
using 
```dart
ZegoUIKitPrebuiltCallInvitationService().init()
```

On the first initialization and permission requests, the Permissions - display over other apps open settings without popup. and  return status.

**Expected behavior:** 
there should be a popup question for the user,and it should be customizable since the list will be different from user to user; it is confusing what to do; a customizable instruction would be nice.

The init function shouldn't be void and should have some return status to know if and debug and future planing if there's something wrong with the initialization.

**Current behavior:** 
- settings opened without user consent.
- no explanation of what to do.
- void return on init function. 

## Steps to reproduce

1. run init()
2. accepts permissions 
3. settings opened without any explanation.

## Images   
as a user i won't be sure what to do here!
<img width="446" alt="Screenshot 2024-08-14 at 5 17 01 AM" src="https://github.com/user-attachments/assets/d80dc109-2773-4c9e-8bbb-bd7e9c5544a2">

**Comment by yoer** [comment]
Do you mean to add a permission confirmation box? 
Will this view pop up after confirmation?

**Comment by moesaid** [comment]
> Do you mean to add a permission confirmation box?

Yeah, a pop-up with a confirmation box like any other native permission pop-up. Just to keep a close native experience and to let the user know what to do. 

> Will this view pop up after confirmation?

no, it should be before going to the settings to simulate a native experience 

------------ 

A type return instead of the void to know the status of the initialization and the permissions grants status would be nice!

**Comment by yoer** [comment]
zego_uikit_prebuilt_call: ^4.15.0
have added permission request pop-up box for android systemAlertWindow

**Comment by nikhil-raj248** [comment]
hi @moesaid , can u pls check this https://github.com/ZEGOCLOUD/zego_uikit_prebuilt_call_flutter/issues/80 , im encountering 2 isuues 
a) Everytime its asking for camera and microphone permission check [this](https://github.com/ZEGOCLOUD/zego_uikit_prebuilt_call_flutter/issues/80#issuecomment-2657520894)
b) app not opening from background state without display over other app (in same thread)

Did u face similar issues in ZegoUIKitPrebuiltCallInvitationService().init
